prow/git/v2/client_factory.go

```diff
@@ -397,13 +377,23 @@ func (c *clientFactory) ensureFreshPrimary(cacheDir string, cacheClientCacher ca
 		// This is the default behavior if FetchCommits is empty or nil (i.e.,
 		// when we don't define a targeted list of commits to fetch directly).
 		if repoOpts.FetchCommits.Len() == 0 {
+			defer repoLock.Unlock()
 			if err := cacheClientCacher.RemoteUpdate(); err != nil {
```

Comment:
Since RemoteUpdate() uses "--prune", probably should document on RepoOpts.ShareObjectsWithSourceRepo that it must be used only in conjunction with FetchCommits. (This is preexisting behavior, but important to understand for future use/review of these features.)

Reply:
Done (also added a check to enforce this rule before we create the primary/secondary clones).